packages/myst-frontmatter/src/numbering/validators.ts

```diff
@@ -3,39 +3,67 @@ import {
   defined,
   incrementOptions,
   validateBoolean,
-  validateObjectKeys,
+  validateKeys,
+  validateObject,
   validateString,
 } from 'simple-validators';
 import type { Numbering } from './types.js';
 
-export const NUMBERING_KEYS = [
-  'enumerator',
-  'figure',
-  'equation',
-  'table',
-  'code',
-  'heading_1',
-  'heading_2',
-  'heading_3',
-  'heading_4',
-  'heading_5',
-  'heading_6',
-];
+export const NUMBERING_OPTIONS = ['enumerator', 'headings'];
+
+const HEADING_KEYS = ['heading_1', 'heading_2', 'heading_3', 'heading_4', 'heading_5', 'heading_6'];
+export const NUMBERING_KEYS = ['figure', 'equation', 'table', 'code', ...HEADING_KEYS];
```

Comment:
And then I guess this just works with the `enumerate` transform, since it already increments count based on `kind`. Nice.

Reply:
Yeah, I was impressed at how much of this already just worked and all we needed to do was to be less strict in validation!